docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,585 @@
+{ "index": {}}
+{ "@timestamp": "2020-04-30T14:31:32-05:00", "message" : "247.37.0.0 - - [2020-04-30T14:31:32-05:00] \"GET /images/nav_bg_top.gif HTTP/1.0\" 200 929"}
+{ "index": {}}
+{ "@timestamp": "2020-04-30T14:31:43-05:00", "message" : "247.37.0.0 - - [2020-04-30T14:31:43-05:00] \"GET /french/images/nav_venue_off.gif HTTP/1.0\" 304 0"}
+----
+// TEST[continued]
+
+At this point, you can view how {es} stores your raw data.
+
+[source,console]
+----
+GET /my-index
+----
+// TEST[continued]
+
+The mapping contains two fields: `@timestamp` and `message`.
+
+[source,console-result]
+----
+{
+  "my-index" : {
+    "aliases" : { },
+    "mappings" : {
+      "properties" : {
+        "@timestamp" : {
+          "type" : "date",
+          "format" : "strict_date_optional_time||epoch_second"
+        },
+        "message" : {
+          "type" : "wildcard"
+        }
+      }
+    },
+    ...
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"aliases": $body.my-index.aliases, "settings": $body.my-index.settings/]
+
+If you want to retrieve results that include `clientip`, you can add that field
+as a runtime field in the mapping. The runtime script operates on the `clientip`
+field at runtime to calculate values for that field.
+
+[source,console]
+----
+PUT /my-index/_mapping
+{
+  "runtime": {
+    "clientip": {
+      "type": "ip",
+      "script" : {
+      "source" : "String m = doc[\"message\"].value; int end = m.indexOf(\" \"); emit(m.substring(0, end));"
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+Using the `clientip` runtime field, you can define a simple query to run a
+search for a specific IP address and return all related fields.
+
+[source,console]
+----
+GET my-index/_search
+{
+  "query": {
+    "match": {
+      "clientip": "211.11.9.0"
+    }
+  },
+  "fields" : ["*"]
+}
+----
+// TEST[continued]
+
+The API returns the following result. Without building your data structure in
+advance, you can search and explore your data in meaningful ways to experiment
+and determine which fields to index.
+
+[source,console-result]
+----
+{
+  ...
+  "hits" : {
+    "total" : {
+      "value" : 2,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "my-index",
+        "_id" : "8Jh81nQBp2DRDXdiOxVt",
+        "_score" : 1.0,
+        "_source" : {
+          "@timestamp" : "2020-06-21T15:00:01-05:00",
+          "message" : "211.11.9.0 - - [2020-06-21T15:00:01-05:00] \"GET /english/index.html HTTP/1.0\" 304 0"
+        },
+        "fields" : {
+          "clientip" : [
+            "211.11.9.0"
+          ],
+          "message" : [
+            "211.11.9.0 - - [2020-06-21T15:00:01-05:00] \"GET /english/index.html HTTP/1.0\" 304 0"
+          ],
+          "@timestamp" : [
+            "2020-06-21T20:00:01.000Z"
+          ]
+        }
+      },
+      {
```

Comment:
I think it'd be best if the example only returned a single line because it'd be a little smaller on the screen and still show everything.

Reply:
@nik9000, do you mean return the same results but not formatted? If so, that means users have to scroll really far to the right, which I think is difficult to read.

Reply:
sorry! I mean a single doc. formatted is just fine. You can certainly play with the formatting to make it smaller. The tests don't mind. It might be worth doing a little of that, but not much. Like, now that I think about it, maybe moving the arrays in `fields` to a single line each. But, yeah, I just meant one result.